src/valuetypes.ts

```diff
@@ -530,20 +542,30 @@ export async function guessValueType(editors: ValueTypeEditor[], name: string, i
   return results[0].editor;
 }
 
-export function createTypeEditor(editors: ValueTypeEditor[], current: ValueTypeEditor, emptyOne = true, idType: string = '') {
-  return `
-  <select class='form-control'>
-          ${emptyOne? '<option value=""></option>':''}
-          ${editors.map((editor) => `<option value="${editor.id}" ${current && current.id === editor.id ? 'selected="selected"' : ''}>${editor.name} ${current && current.id === editor.id && current.name === 'IDType' ? `(${idType})` : ''}</option>`).join('\n')}
-        </select>
-        <span class="input-group-btn">
-          <button class="btn btn-secondary" ${!current || !current.hasEditor ? 'disabled="disabled' : ''} type="button"><i class="glyphicon glyphicon-cog"></i></button>
-        </span>`;
+export function createTypeEditor(editors: ValueTypeEditor[], current: ValueTypeEditor, def: ITypeDefinition, emptyOne = true) {
+  const options = editors.map((editor) => editor.getOptionsMarkup(current, def)).join('\n');
+
+ return `<select class="form-control">
+        ${emptyOne? '<option value=""></option>':''}
+        ${options}
+    </select>
+    <span class="input-group-btn">
+      <button class="btn btn-secondary" ${!current || !current.hasEditor ? 'disabled="disabled' : ''} type="button"><i class="glyphicon glyphicon-cog"></i></button>
+    </span>`;
 }
 
 export function updateType(editors: ValueTypeEditor[], emptyOne = true) {
   return function (d) {
-    const type = ((emptyOne && this.selectedIndex <= 0)) ? null : editors[this.selectedIndex < 0 ? 0 : this.selectedIndex - (emptyOne?1:0)];
+    const parent = this.options[this.selectedIndex].parentNode;
+
+    let type = null;
+    if(parent.nodeName !== 'OPTGROUP') {
```

Comment:
I think it should work as before - as far as I can tell before the editor was also null, when nothing is selected, but I tried it and I it does obviously not work, because the editor is null and later on the parse method is called
would a string editor be the correct one, if no option is selected?